scripts/dev-e2e.sh

```diff
@@ -0,0 +1,176 @@
+#!/bin/bash
+
+GREEN='\033[0;32m'
+NC='\033[0m'
+
+STATUSFILE='armadactl-query.txt'
+AOREPO='https://github.com/armadaproject/armada-operator.git'
+AOHOME=../armada-operator
+ARMADA_CTL_VERSION='0.18.3'
+TEST_QUEUE='test'
+JOB_DETAILS=1
+
+usage() {
+  cat > /dev/stderr  <<USAGE
+Usage: ${0} -s
+  -s => get job status details from Armada
+
+USAGE
+  exit 1
+}
+
+while getopts "s" opt; do
+    case "${opt}" in
+        s )
+          JOB_DETAILS=1 ;;
+        * )
+          echo "opt is $opt"
+          usage ;;
+    esac
+done
+
+scripts="$(cd "$(dirname "$0")" || exit; pwd)"
+source "$scripts"/init.sh
+
+log() {
+    echo -e "${GREEN}$1${NC}"
+}
+
+
+fetch-armadactl() {
+  os=$(uname -s)
+  arch=$(uname -m)
+  dl_url='https://github.com/armadaproject/armada/releases/download'
+
+  if [ "$os" = "Darwin" ]; then
+    dl_os='darwin'
+  elif [ "$os" = "Linux" ]; then
+    dl_os='linux'
+ else
+    log "fetch-armadactl(): sorry, operating system $os not supported; exiting now"
+    exit 1
+  fi
+
+  if [ "$arch" = 'arm64' ]; then
+    dl_arch='arm64'
+  elif [ "$arch" = 'x86_64' ]; then
+    dl_arch='amd64'
+  else
+    log "fetch-armadactl(): sorry, architecture $arch not supported; exiting now"
+    exit 1
+  fi
+
+  dl_file=armadactl_${ARMADA_CTL_VERSION}_${dl_os}_${dl_arch}.tar.gz
+
+  curl -s -LO ${dl_url}/v${ARMADA_CTL_VERSION}/$dl_file
+  tar xzf $dl_file armadactl
+}
+
+armadactl-retry() {
+  for _ in {1..10}; do
+    if ./armadactl "$@" > /dev/null 2>&1; then
+      return 0
+    fi
+    sleep 5
+  done
+  log "armadactl command failed after 10 attempts" >&2
+  exit 1
+}
```

Comment:
```suggestion
armadactl-retry() {
  for attempt in {1..10}; do
    if ./armadactl "$@" > /dev/null 2>&1; then
      return 0
    fi
    sleep 5
  done
  err "armadactl command failed after $attempt attempts"
  exit 1
}
```

Reply:
Good suggestion - fixed in commit 1bc04a0.